lib/portlayer/attach/communication/connector.go

```diff
@@ -127,23 +129,82 @@ func (c *Connector) Get(ctx context.Context, id string, timeout time.Duration) (
 	}
 }
 
-func (c *Connector) Remove(id string) error {
+// Remove removes the session the inteactions map
+func (c *Connector) RemoveInteraction(id string) error {
 	defer trace.End(trace.Begin(id))
 
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
 	var err error
 
-	if c.connections[id] != nil {
-		if c.connections[id].id == id {
-			err = c.connections[id].spty.Close()
-		}
-		delete(c.connections, id)
+	if c.interactions[id] != nil {
+		log.Debugf("Removing %s from the map", id)
+		err = c.interactions[id].Close()
+		delete(c.interactions, id)
+	}
+
+	for i := range c.interactions {
+		log.Debugf("We have %s in the map after the RemoveInteraction", c.interactions[i])
 	}
 	return err
 }
 
+// Start starts the connector
+func (c *Connector) Start() {
+	defer trace.End(trace.Begin(""))
+
+	c.wg.Add(1)
+	go c.serve()
+}
+
+// Stop stops the connector
+func (c *Connector) Stop() {
+	defer trace.End(trace.Begin(""))
+
+	c.listener.Close()
+	close(c.done)
+	c.wg.Wait()
+}
+
+// URL returns the listener's URL
+func (c *Connector) URL() string {
```

Comment:
@caglar10ur are you talking about this?
https://github.com/vmware/vic/blob/master/lib/vspc/helpers.go#L85